src/modules/tools/zprobe/CartGridStrategy.cpp

```diff
@@ -229,41 +287,41 @@ bool CartGridStrategy::load_grid(StreamOutput *stream)
         return false;
     }
 
-    for (int y = 0; y < grid_size; y++) {
-        for (int x = 0; x < grid_size; x++) {
-            if(fread(&grid[x + (grid_size * y)], sizeof(float), 1, fp) != 1) {
+    for (int y = 0; y < configured_grid_y_size; y++) {
+        for (int x = 0; x < configured_grid_x_size; x++) {
+            if(fread(&grid[x + (configured_grid_x_size * y)], sizeof(float), 1, fp) != 1) {
                 stream->printf("error:Failed to read grid\n");
                 fclose(fp);
                 return false;
             }
         }
     }
-    stream->printf("grid loaded, grid: (%f, %f), size: %d\n", x_size, y_size, grid_size);
+    stream->printf("grid loaded, grid: (%f, %f), size: %d x %d\n", x_size, y_size, load_grid_x_size, load_grid_y_size);
     fclose(fp);
     return true;
 }
 
-bool CartGridStrategy::probe_grid(int n, float x_size, float y_size, StreamOutput *stream)
+bool CartGridStrategy::probe_grid(int n, int m, float x_start, float y_start, float x_size, float y_size, StreamOutput *stream)
 {
-    if(n < 5) {
+    if((n < 5)||(m < 5)) {
         stream->printf("Need at least a 5x5 grid to probe\n");
         return true;
     }
 
     float initial_z = findBed();
     if(isnan(initial_z)) return false;
 
-    float x_step= x_size / n;
-    float y_step= y_size / n;
+    float x_step = x_size / n;
+    float y_step = y_size / m;
     for (int c = 0; c < n; ++c) {
-        float y = y_step * c;
+        float y = y_start + y_step * c;
         for (int r = 0; r < n; ++r) {
-            float x = x_step * r;
+            float x = x_start + x_step * r;
             float z = 0.0F;
             float mm;
             if(!zprobe->doProbeAt(mm, x, y)) return false;
             z = zprobe->getProbeHeight() - mm;
-            stream->printf("%8.4f ", z);
+            stream->printf("%1.4f ", z);
```

Comment:
I wanted it left as 1 which was the old way it was. this means that when it is printed there are no preceding spaces, but does not limit the pre period length.